litestar/middleware/_internal/cors.py

```diff
@@ -113,7 +113,13 @@ def _create_preflight_response(self, origin: str, request_headers: Headers) -> R
                 response_headers["Access-Control-Allow-Headers"] = ", ".join(
                     sorted(set(pre_flight_requested_headers) | DEFAULT_ALLOWED_CORS_HEADERS)  # pyright: ignore
                 )
-            elif any(header.lower() not in self.config.allow_headers for header in pre_flight_requested_headers):
+            elif any(
+                header.lower()
+                not in set(self.config.allow_headers).union(
+                    default_header.lower() for default_header in DEFAULT_ALLOWED_CORS_HEADERS
```

Comment:
The construction of this set should be moved out of the loop. Otherwise it would be re-created on every iteration. Probably make an `else` branch out of the `elif` so its construction is still conditional.